In [6]:
#import stuff
import pandas as p
import numpy as n
import seaborn as s
from matplotlib import pyplot
import lib as l

In [9]:
#import data
pokemon = p.read_csv('data/pokemon.csv', header = None, 
                     skiprows=1, encoding = 'latin-1', low_memory = False,
                    names = ['Name', 'Version', 'Primary Type' ,'Secondary Type', 'Attack', 'Defense' , 'HP', 'Sp. Attack', 'Sp. Defense', 'Speed', 'Total']
)
pokemon

Name        Version Primary Type Secondary Type  Attack  Defense  \
0      Bulbasaur            NaN        GRASS         POISON      49       49   
1        Ivysaur            NaN        GRASS         POISON      62       63   
2       Venusaur            NaN        GRASS         POISON      82       83   
3       Venusaur  Mega Venusaur        GRASS         POISON     100      123   
4     Charmander            NaN         FIRE            NaN      52       43   
...          ...            ...          ...            ...     ...      ...   
1040   Glastrier            NaN          ICE            NaN     145      130   
1041   Spectrier            NaN        GHOST            NaN      65       60   
1042     Calyrex            NaN      PSYCHIC          GRASS      80       80   
1043     Calyrex      Ice Rider      PSYCHIC            ICE     165      150   
1044     Calyrex   Shadow Rider      PSYCHIC          GHOST      85       80   

       HP  Sp. Attack  Sp. Defense  Speed  Total  
0      45          65           65     45    318  
1      60          80           80     60    405  
2      80         100          100     80    525  
3      80         122          120     80    625  
4      39          60           50     65    309  
...   ...         ...          ...    ...    ...  
1040  100          65          110     30    580  
1041  100         145           80    130    580  
1042  100          80           80     80    500  
1043  100          85          130     50    680  
1044  100         165          100    150    680  

[1045 rows x 11 columns]

# Question 1;

This is the data set of all current pokemon as of 11/23/21. This represents a statistical outlook of all 1045 pokemon. This includes legendaries, mythical, and mega evolution variants. Their attributes are broken down into 7 numerical categories. They represents how strong the corresponding attribute of the pokemon is. So the higher the value, the stronger the attribute will be. The categories are as follows:

    - HP
    - Attack
    - Defense
    - Special Attack
    - Special Defense
    - Speed
    - Total
    
These are not the actual values of the attributes. _i.e._ If I have a Lvl 100 Venusaur, I won't have 82 Attack. It is more like a scale that determines the boundaries that a pokemon's attributes are tied to. 

# Question 2;

There are few, not so obvious **classes** that are present in this data set. One of which can be determining if a pokemon is considered _Statistically Strong_. This has a vague definition, but generally the higher the pokemon's attributes are, the better they can do in battle. 

# K-nearest Neighbors Algorithm

# K-means Clustering